.github/workflows/crystal.yml

```diff
@@ -2,31 +2,57 @@ name: CI
 
 on:
   push:
-  workflow_dispatch:
   schedule:
     - cron: "0 6 * * 1"
 
 jobs:
-  build:
+  style:
+    runs-on: ubuntu-latest
+    container: crystallang/crystal
+    steps:
+      - uses: actions/checkout@v2
+      - name: Format
+        run: crystal tool format --check
+      - name: Lint
+        uses: crystal-ameba/github-action@v0.2.12
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  test:
     strategy:
       fail-fast: false
       matrix:
-        experimental: [false]
-        os: [ubuntu-latest]
+        stable: [true]
         crystal:
-          - latest
+          - 1.1.0
           - 1.0.0
           - 0.36.1
         include:
-          - experimental: true
-            os: ubuntu-latest
+          - stable: false
             crystal: nightly
-    runs-on: ${{ matrix.os }}
-    continue-on-error: ${{ matrix.experimental }}
+    name: "crystal: ${{ matrix.crystal }}, stable: ${{ matrix.stable }}"
+    runs-on: ubuntu-latest
     container: crystallang/crystal:${{ matrix.crystal }}-alpine
```

Comment:
It would be good to use https://github.com/crystal-lang/install-crystal rather than containers for this.

This will then annotate compilation and spec issues inline.